product_supplierinfo_for_customer_sale_quick/models/product_product.py

```diff
@@ -0,0 +1,38 @@
+from odoo import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    product_customer_code = fields.Char(
+        string="Product Customer Code",
+        compute="_compute_product_customer_info",
+        search="_search_customer_supplierinfo_data",
+        readonly=True,
+    )
+
+    product_customer_name = fields.Char(
+        string="Product Customer Name",
+        compute="_compute_product_customer_info",
+        search="_search_customer_supplierinfo_data",
+        readonly=True,
+    )
+
+    @api.model
+    def _search_customer_supplierinfo_data(self, operator, value):
+        products = self.env["product.product"].search(
+            [
+                "|",
+                ("product_customer_code", operator, value),
+                ("product_customer_name", operator, value),
+            ]
+        )
+        return [("id", "in", products.ids)]
+
+    def _compute_product_customer_info(self):
+        for product in self:
+            partner_id = product.env.context.get("partner_id")
+            partner = self.env["res.partner"].browse(partner_id)
+            supplier_info = product._select_customerinfo(partner=partner)
+            product.product_customer_code = supplier_info.product_code
+            product.product_customer_name = supplier_info.product_name
```

Comment:
**todo**: We need to make sure this function always assigns a value. But if there's no `partner` we can avoid passing from `_select_customerinfo` and immediately assign `False` to both fields.